cmds/import_image/import_image_cmd.go

```diff
@@ -84,5 +114,28 @@ func (cmd *ImportImageCmd) Run() error {
 	cmd.Id = vgbdtgObject.Id
 	cmd.Uuid = vgbdtgObject.GlobalIdentifier
 
+	if cmd.Public {
+		execStmtRetryable := boshretry.NewRetryable(
+			func() (bool, error) {
+				id, err := vgbdtgService.CreatePublicArchiveTransaction(cmd.Id, cmd.PublicName, cmd.PublicNote, cmd.PublicNote, locations)
+				if err != nil {
+					return true, errors.New(fmt.Sprintf("There would be an active transaction in progress."))
+				}
+
+				cmd.Id = id
```

Comment:
Add a issue in `softlayer-go` for `getGlobalIdentifier` and I will address it when I have a minute.